### PR TITLE
Usage statuses #486

### LINKED ISSUE #486: usage statuses
- Add endpoint
- remove enum

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

- remove usage statuses enum and replace with list retrieved from the backend 
- Integrated new usage statuses endpoint into items
## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

closes #486

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
This pr needs to wait until the back-end one is merged in. To get the e2e with api to work

File: src/items/itemDialog.component.tsx
Comment (joelvdavies) [inline src/items/itemDialog.component.tsx:759]: Is it possible to make this show all the time - on a slow connection it adds to the form after a while which I think is also an accessibility issue, it should be there but just not display any options until loaded.

**Comment by MatteoGuarnaccia5** [comment]
api e2e tests pass locally. requires [#253 to be merged in on the backend](https://github.com/ral-facilities/inventory-management-system-api/pull/253)

**Comment by MatteoGuarnaccia5** [comment]
@joshuadkitenge I can't request your review as it is your PR but you should probably look at it as well

**Review by joelvdavies — APPROVED** [review]
I approve to merge into https://github.com/ral-facilities/inventory-management-system/pull/544. Full testing can then be done there.

**Pull request merged**
